.github/workflows/build_and_test_dbt.yaml

```diff
@@ -0,0 +1,122 @@
+name: Build and test dbt
+
+on:
+  pull_request:
+    branches: [master, data-catalog]
+    # Specifying event types manually allows us to run this flow when the
+    # PR is closed so that we can clean up staging dbt resources
+    types:
+      - opened
+      - synchronize
+      - closed
+      - reopened
```

Comment:
Note that these event types represent the default event types plus `closed`. [Docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request):

> By default, a workflow only runs when a pull_request event's activity type is opened, synchronize, or reopened.